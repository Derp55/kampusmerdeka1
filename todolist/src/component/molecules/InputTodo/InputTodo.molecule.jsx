import React from "react";
import "./InputTodo.molecule.style.css";

const InputTodo = ({ value, onChange, onSubmit }) => {
  return (
    <div className="input-todo">
      <input
        type="text"
        className="input-todo__field"
        value={value}
        onChange={onChange}
        placeholder="Input todo . . ."
      />
      <button className="input-todo__button" onClick={onSubmit}>
        Submit
      </button>
    </div>
  );
};

export default InputTodo;
