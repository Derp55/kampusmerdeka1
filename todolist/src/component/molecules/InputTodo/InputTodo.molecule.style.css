.input-todo {
  width: 400px;
  margin: 20px 0;
  display: flex;
}

.input-todo .input-todo__field {
  width: 100%;
  flex: 1;
  padding: 10px;
}
