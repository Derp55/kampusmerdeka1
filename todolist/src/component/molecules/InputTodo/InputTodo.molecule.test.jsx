import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import InputTodo from "./InputTodo.molecule";
describe("InputTodo.molecule.jsx", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<InputTodo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correct value if isEdit false", () => {
    render(
      <InputTodo
        isEdit={false}
        onChange={vi.fn()}
        onSubmit={vi.fn()}
        onUpdate={vi.fn()}
        value="Halo dunia"
      />
    );

    expect(screen.getByText("Submit")).toBeInTheDocument();
  });

  it("should render correct value if isEdit true", () => {
    render(
      <InputTodo
        isEdit={true}
        onChange={vi.fn()}
        onSubmit={vi.fn()}
        onUpdate={vi.fn()}
        value="Halo dunia"
      />
    );

    expect(screen.getByText("Update")).toBeInTheDocument();
  });
});
