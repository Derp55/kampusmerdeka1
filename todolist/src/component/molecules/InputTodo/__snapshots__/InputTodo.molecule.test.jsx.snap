// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputTodo.molecule.jsx > should render correctly 1`] = `
<div
  className="input-todo"
  id="input-todo"
>
  <input
    className="input-todo__field"
    placeholder="Input todo . . ."
    type="text"
  />
  <button
    className="input-todo__button"
  >
    Submit
  </button>
</div>
`;
